### PR TITLE
Remove cardano-wallet-http-bridge

### PR DESCRIPTION
**Pull request by Anviking**
#897 

# Overview
- [x] I have removed `cardano-wallet-http-bridge`
- [x] I have adjusted some comments referring to http-bridge
- [x] I have removed the `restore` benchmark.
- [x] I've rebased and re-opened this PR.

# Comments
- I'd like the CI to run, so I won't make this a draft PR.

### CODE REVIEW

File: release.nix [hunk 1]
```diff
@@ -31,8 +31,6 @@ let
       # fixme: fix failing tests
       # collectTests jobs.native.tests ++
       collectTests jobs.native.benchmarks ++
-      [ jobs.native.cardano-wallet-http-bridge.x86_64-linux
-        jobs.native.cardano-wallet-http-bridge.x86_64-darwin
         jobs.native.cardano-wallet-jormungandr.x86_64-linux
```
Comment (paweljakubas) [inline release.nix]: don't we need here `[` ???

**Comment by paweljakubas** [comment]
So we removed `.buildkite/bench-restore.sh ` which used `http-bridge` but cannot use `jormungandr` at this moment. And we will resurrect this stuff when `haskell-node` can make it possible, right?

**Comment by piotr-iohk** [comment]
>So we removed .buildkite/bench-restore.sh which used http-bridge but cannot use jormungandr at this moment. And we will resurrect this stuff when haskell-node can make it possible, right?

It would be nice actually to find a way to have some benchmarks in `jormungandr`, especially that there is a rollback story comming up and it has some performance requirements in it (memory usage, sync time and disk space)

**Comment by paweljakubas** [comment]
@piotr-iohk  definitely, but I do think adding those benchmarks in not the subject of this PR. The only question for me is whether we retain `.buildkite/bench-restore.sh` in some dummy form or not

File: lib/core/test/manual/Cardano/Launcher/POSIXSpec.md [hunk 2]
```diff
@@ -21,7 +21,7 @@ signals are correctly handled on a unix system.
   ```
   $ ps -ef | grep cardano
   10983  4904  0 09:42 pts/1 00:00:00 /home/user/Documents/IOHK/cardano-wallet/.stack-work/install/x86_64-linux/lts-13.8/8.6.3/bin/cardano-wallet
-  11003 10983 75 09:42 pts/1 00:03:30 cardano-http-bridge start --port 8080
+  11003 10983 75 09:42 pts/1 00:03:30 cardano-wallet-jormungandr start --port 8080
```
Comment (paweljakubas) [inline lib/core/test/manual/Cardano/Launcher/POSIXSpec.md]: @Anviking we are sure here that we can replace `http-bridge` with `wallet-jormungandr` ? (I mean all this file)

File: nix/.stack.nix/cardano-wallet-jormungandr.nix [hunk 3]
```diff
@@ -83,7 +83,6 @@
             (hsPkgs.cardano-wallet-cli)
             (hsPkgs.cardano-wallet-core)
             (hsPkgs.cardano-wallet-core-integration)
-            (hsPkgs.cardano-wallet-http-bridge)
```
Comment (paweljakubas) [inline nix/.stack.nix/cardano-wallet-jormungandr.nix]: side note: that's surprising for me (I mean that it was here)

**Review by paweljakubas — APPROVED** [review]
LGTM! I was thinking about leaving `.buildkite/bench-restore.sh` with some dummy target value, but as `http-bridge` was only target valid here I am more inclined to remove it for time being (as it was done), and resurrect it when needed (presumably when restoration benchmark is to be supported for jormungandr or cardano-node aka haskell node). On the other hand, maybe we should postpone this until we have something interacting properly (jormungandr cannot) with byron mainnet.

So, from technical point of view nuking of `http-bridge` was done as it should be. But, should we have this discontinuity of servicing byron mainnet? On the other hand, byron era is coming to an end and we are supposed to support new era, so maybe we should not be here very strict. http-bridge served us well for several months, helped to pluck out a number of bugs in core, but requires some attention/effort/resources that could be dedicated solely on jormungandr and cardano-node - I would nuke it in this form continue our stories without http-bridge burden, and when finished start story which establishes restoration using jormungandr/cardano-node backend

**Review by rvl — CHANGES_REQUESTED** [review]
It's good that you have scoped out all the work for this, but if it means that we no longer have any wallet that's capable of connecting to byron mainnet, then Nack.

**Comment by piotr-iohk** [comment]
>It's good that you have scoped out all the work for this, but if it means that we no longer have any wallet that's capable of connecting to byron mainnet, then Nack.

I do have some reservations with regards to not having benchmarks mainnet any longer, on the other hand the support for `http-bridge` was only provisional - just to have a chain to work with (cause _official_ wallet backend for the byron era is the one from `cardano-sl`)... 

I think the question is if leaving support for `http-bridge` for the sake of having benchmarks on mainnet is worth of additional maintenance effort. And are those benchmarks really adding value knowing that `http-bridge` will be eventually dropped. :thinking:

File: lib/core/test/manual/Cardano/Launcher/POSIXSpec.md (hunk 2 above)
Comment (Anviking) [reply lib/core/test/manual/Cardano/Launcher/POSIXSpec.md]: I replaced the commands in the guide carefully, verifying that they worked.

But my `ps -ef | grep cardano` output was a lot messier, so I didn't copy-paste, but rather just tweak the logs. The log-output-examples are therefore marginally incorrect.

c5cf30c

**Comment by Anviking** [comment]
Closing for now.

**Comment by Anviking** [comment]
No real references to the bridge remaining:
```bash
$ git grep -1 "bridge"
lib/core-integration/src/Test/Integration/Scenario/CLI/Miscellaneous.hs-                , "serve --network testnet --random-por --quiet"
lib/core-integration/src/Test/Integration/Scenario/CLI/Miscellaneous.hs:                , "serve --network testnet --random-port --bridge-port 666 --vebrose"
lib/core-integration/src/Test/Integration/Scenario/CLI/Miscellaneous.hs-                , "serve --network"
--
lib/core/src/Cardano/Byron/Codec/Cbor.hs--- of Cardano and will endure in the first stages of Shelley. They are also used
lib/core/src/Cardano/Byron/Codec/Cbor.hs:-- by components like the Rust <https://github.com/input-output-hk/cardano-http-bridge cardano-http-bridge>.
lib/core/src/Cardano/Byron/Codec/Cbor.hs-
--
lib/core/src/Cardano/Byron/Codec/Cbor.hs-        , blockHeight = Quantity $ fromIntegral difficulty
lib/core/src/Cardano/Byron/Codec/Cbor.hs:        , headerHash = Hash "http-bridge"
lib/core/src/Cardano/Byron/Codec/Cbor.hs-        , parentHeaderHash = previous
--
lib/core/src/Cardano/Byron/Codec/Cbor.hs-        , blockHeight = bh
lib/core/src/Cardano/Byron/Codec/Cbor.hs:        , headerHash = Hash "http-bridge"
lib/core/src/Cardano/Byron/Codec/Cbor.hs-        , parentHeaderHash = previous
--
lib/core/test/unit/Cardano/Byron/Codec/CborSpec.hs-    , blockHeight = Quantity 2276029
lib/core/test/unit/Cardano/Byron/Codec/CborSpec.hs:    , headerHash = Hash "http-bridge"
lib/core/test/unit/Cardano/Byron/Codec/CborSpec.hs-    , parentHeaderHash = Hash $ unsafeFromHex
--
lib/core/test/unit/Cardano/Byron/Codec/CborSpec.hs-        , blockHeight = Quantity 2276028
lib/core/test/unit/Cardano/Byron/Codec/CborSpec.hs:        , headerHash = Hash "http-bridge"
lib/core/test/unit/Cardano/Byron/Codec/CborSpec.hs-        , parentHeaderHash = parentHeaderHash0
--
lib/core/test/unit/Cardano/Byron/Codec/CborSpec.hs-        , blockHeight = Quantity 2276385
lib/core/test/unit/Cardano/Byron/Codec/CborSpec.hs:        , headerHash = Hash "http-bridge"
lib/core/test/unit/Cardano/Byron/Codec/CborSpec.hs-        , parentHeaderHash = parentHeaderHash0
--
lib/core/test/unit/Cardano/Byron/Codec/CborSpec.hs-        , blockHeight = Quantity 657163
lib/core/test/unit/Cardano/Byron/Codec/CborSpec.hs:        , headerHash = Hash "http-bridge"
lib/core/test/unit/Cardano/Byron/Codec/CborSpec.hs-        , parentHeaderHash = parentHeaderHash0
--
lib/core/test/unit/Cardano/Byron/Codec/CborSpec.hs-        , blockHeight = Quantity 302376
lib/core/test/unit/Cardano/Byron/Codec/CborSpec.hs:        , headerHash = Hash "http-bridge"
lib/core/test/unit/Cardano/Byron/Codec/CborSpec.hs-        , parentHeaderHash = parentHeaderHash0
--
lib/jormungandr/src/Cardano/Wallet/Jormungandr/Transaction.hs-        -- without the witnesses (since we don't yet have them!). In this sense,
lib/jormungandr/src/Cardano/Wallet/Jormungandr/Transaction.hs:        -- this is a transaction id as Byron nodes or the http-bridge
lib/jormungandr/src/Cardano/Wallet/Jormungandr/Transaction.hs-        -- defines them.
--
lib/jormungandr/test/data/jormungandr/genesis.yaml-# test's faucets to be of similar shape with the one used with the now removed
lib/jormungandr/test/data/jormungandr/genesis.yaml:# http-bridge, we want each wallet to have 10 UTxO available, hence the
lib/jormungandr/test/data/jormungandr/genesis.yaml-# repetition for each address.
--
lib/jormungandr/test/integration/Cardano/Faucet.hs-      , "garage", "husband", "wreck", "win", "used"
lib/jormungandr/test/integration/Cardano/Faucet.hs:      , "acid", "seed", "casual", "bridge", "gallery"
lib/jormungandr/test/integration/Cardano/Faucet.hs-      ]
--
lib/jormungandr/test/integration/Main.hs-
lib/jormungandr/test/integration/Main.hs:-- | Define the actual executable name for the bridge CLI
lib/jormungandr/test/integration/Main.hs-instance KnownCommand (Jormungandr n) where
--
lib/jormungandr/test/integration/Main.hs-        -- We use a range (min, max) and call it an "estimator" because for the
lib/jormungandr/test/integration/Main.hs:        -- bridge (and probably cardano-node on Shelley), it's not possible to
lib/jormungandr/test/integration/Main.hs-        -- compute the fee precisely by only knowing the number of inputs and
```

File: cardano-wallet.cabal [hunk 4]
```diff
@@ -0,0 +1,60 @@
+name:                cardano-wallet
```
Comment (KtorZ) [inline cardano-wallet.cabal]: This file shouldn't exist. We moved executables into their corresponding packages.

File: lib/core/test/manual/Cardano/Launcher/POSIXSpec.md [hunk 5]
```diff
@@ -32,14 +32,14 @@ signals are correctly handled on a unix system.
 
 ### `SIGQUIT`
 
-- Start the launcher using `stack exec -- cardano-wallet launch`
+- Start the launcher using `stack exec -- cardano-wallet-jormungandr launch --genesis-block lib/jormungandr/test/data/jormungandr/block0.bin --bft-leaders lib/jormungandr/test/data/jormungandr/secret.yaml`
```
Comment (KtorZ) [inline lib/core/test/manual/Cardano/Launcher/POSIXSpec.md]: `--bft-leaders` doesn't exist anymore. This should be updated to reflect the latest version :+1:

File: lib/core/test/manual/Cardano/Launcher/POSIXSpec.md [hunk 6]
```diff
@@ -50,14 +50,14 @@ signals are correctly handled on a unix system.
 
 ### `SIGTERM`
 
-- Start the launcher in background using `stack exec -- cardano-wallet launch &`
+- Start the launcher in background using `stack exec -- cardano-wallet-jormungandr launch --genesis-block lib/jormungandr/test/data/jormungandr/block0.bin --bft-leaders lib/jormungandr/test/data/jormungandr/secret.yaml &`
```
Comment (KtorZ) [inline lib/core/test/manual/Cardano/Launcher/POSIXSpec.md]: `--bft-leaders` doesn't exist anymore. This should be updated to reflect the latest version :+1:

File: lib/core/test/manual/Cardano/Launcher/POSIXSpec.md [hunk 7]
```diff
@@ -32,14 +32,14 @@ signals are correctly handled on a unix system.
...
-- Start the launcher using `stack exec -- cardano-wallet launch`
+- Start the launcher using `stack exec -- cardano-wallet-jormungandr launch --genesis-block lib/jormungandr/test/data/jormungandr/block0.bin --bft-leaders lib/jormungandr/test/data/jormungandr/secret.yaml`
 - In another terminal, run `ps -ef | grep cardano`, there should be three
   processes running (the launcher, the wallet and the underlying chain
   producer)
   ```
   $ ps -ef | grep cardano
   10983  4904  0 09:42 pts/1 00:00:00 /home/user/Documents/IOHK/cardano-wallet/.stack-work/install/x86_64-linux/lts-13.8/8.6.3/bin/cardano-wallet
-  11003 10983 75 09:42 pts/1 00:03:30 cardano-http-bridge start --port 8080
+  11003 10983 75 09:42 pts/1 00:03:30 wallet-jormungandr start --port 8080
```
Comment (KtorZ) [inline lib/core/test/manual/Cardano/Launcher/POSIXSpec.md]: This is wrong. `cardano-http-bridge` referred to the actual bridge application. Here we should be referring to `jormungandr` instead.

File: lib/core/test/manual/Cardano/Launcher/POSIXSpec.md [hunk 8]
```diff
@@ -50,14 +50,14 @@ signals are correctly handled on a unix system.
...
-- Start the launcher in background using `stack exec -- cardano-wallet launch &`
+- Start the launcher in background using `stack exec -- cardano-wallet-jormungandr launch --genesis-block lib/jormungandr/test/data/jormungandr/block0.bin --bft-leaders lib/jormungandr/test/data/jormungandr/secret.yaml &`
 - Run `ps -ef | grep cardano` and lookup the `pid` of the launcher process (and
   control that there are indeed three processes running: the launcher, the
   wallet and the chain producer)
   ```
   $ ps -ef | grep cardano
   10983  4904  0 09:42 pts/1 00:00:00 /home/user/Documents/IOHK/cardano-wallet/.stack-work/install/x86_64-linux/lts-13.8/8.6.3/bin/cardano-wallet
-  11003 10983 75 09:42 pts/1 00:03:30 cardano-http-bridge start --port 8080
+  11003 10983 75 09:42 pts/1 00:03:30 cardano-wallet-jormungandr start --port 8080
```
Comment (KtorZ) [inline lib/core/test/manual/Cardano/Launcher/POSIXSpec.md]: This is wrong. `cardano-http-bridge` referred to the actual bridge application. Here we should be referring to `jormungandr` instead.

File: lib/jormungandr/test/data/jormungandr/genesis.yaml [hunk 9]
```diff
@@ -58,9 +58,9 @@ blockchain_configuration:
 # The initial deposits present in the blockchain.
 #
 # The last 10 addresses come from 10 different wallets. For the integration
-# test's faucets to be of similar shape with the one used with the http-bridge,
-# we want each wallet to have 10 UTxO available, hence the repetition for each
-# address.
+# test's faucets to be of similar shape with the one used with the now removed
+# http-bridge, we want each wallet to have 10 UTxO available, hence the
+# repetition for each address.
```
Comment (KtorZ) [inline lib/jormungandr/test/data/jormungandr/genesis.yaml:63]: I'd actually re-write the comment to be somewhat meaningful :/

**Review by KtorZ — CHANGES_REQUESTED** [review]
Good job cleaning this up. A few remarks, especially in the POSIX.md manual tests which I think should be carefully reviewed.

**Comment by Anviking** [comment]
I did some semi-drastic changes to POSIXSpec.md. 
b7184cf

**Review by KtorZ — APPROVED** [review]

**Review by rvl — APPROVED** [review]

**Review by paweljakubas — APPROVED** [review]
Great! I was expecting this PR for a quarter or more :-)

**Review by piotr-iohk — APPROVED** [review]
:ok_hand:

**Pull request merged**